File: nitpickers/pyql/pyql/gui/widgets/widget_factory.py
from pyql.gui.widgets.widgets import *
from util.types import *
from util.multimethods import multimethod


class WidgetFactory:

    def __init__(self):
        pass

    def widget(self, root, identifier, type, value):
        w = self.create_widget(type)
        return w(root, identifier, value)

    @multimethod(String)
    def create_widget(self, type):
        return TextWidget

    @multimethod(Integer)
    def create_widget(self, type):
        return NumberWidget

    @multimethod(Decimal)
    def create_widget(self, type):
        return DecimalWidget

    @multimethod(Money)
    def create_widget(self, type):
        return MoneyWidget

    @multimethod(Boolean)
    def create_widget(self, type):
        return CheckWidget

File: Abel_Elias/src/gui/widgets/WidgetFactory.java
package gui.widgets;

import QL.classes.values.BooleanValue;
import QL.classes.values.DateValue;
import QL.classes.values.NumericValue;
import QL.classes.values.StringValue;
import QL.classes.values.Value;

public class WidgetFactory {
    public static Widget getDefaultWidget(Value value) {
        switch (value.getType()) {
            case Value.STRING:
                return new TextWidget((StringValue) value);
            case Value.BOOLEAN:
                return new CheckBoxWidget((BooleanValue) value);
            case Value.DATE:
                return new DateWidget((DateValue) value);
            case Value.DECIMAL:
            case Value.MONEY:
            case Value.INTEGER:
                return new TextWidget((NumericValue) value);
            default: break;
        }
        return null;
    }
}
